src/main/java/uk/gov/hmcts/ccd/domain/types/TextCaseReferenceCaseLinkValidator.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.hmcts.ccd.domain.types;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.springframework.beans.factory.annotation.Qualifier;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseFieldDefinition;
+import uk.gov.hmcts.ccd.domain.service.common.CaseService;
+import uk.gov.hmcts.ccd.endpoint.exceptions.ResourceNotFoundException;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.util.Collections;
+import java.util.List;
+
+@Named("TextCaseReferenceCaseLinkValidator")
+@Singleton
+public class TextCaseReferenceCaseLinkValidator implements PredefinedTypeFieldValidator {
+
+    private CaseService caseService;
+    private TextValidator textValidator;
+
+    @Inject
+    public TextCaseReferenceCaseLinkValidator(@Qualifier("TextValidator") TextValidator textValidator, CaseService caseService) {
+        this.caseService = caseService;
+        this.textValidator = textValidator;
+    }
+
+    @Override
+    public List<ValidationResult> validate(final String dataFieldId,
+                                           final JsonNode dataValue,
+                                           final CaseFieldDefinition caseFieldDefinition) {
+
+        List<ValidationResult> validationResults = textValidator.validate(dataFieldId, dataValue, caseFieldDefinition);
+        if (validationResults.isEmpty() && !textValidator.isNullOrEmpty(dataValue)) {
+            final String value = dataValue.textValue();
+            return isAnExistingCase(value, dataFieldId);
+        }
+        return validationResults;
+    }
+
+    private List<ValidationResult> isAnExistingCase(final String value, final String dataFieldId) {
+        try {
+            this.caseService.getCaseDetailsByCaseReference(formatCaseReference(value));
+            return Collections.emptyList();
+        } catch (ResourceNotFoundException resourceNotFoundException) {
+            return Collections.singletonList(
+                new ValidationResult(
+                    value + " does not correspond to an existing CCD case. Please update before proceeding",
```

Comment:
`Please update before proceeding` is more a UI type of error message. But this validator is supposed to be generic. I would remove that from the message

Reply:
what is would be the final message mario ? the current one without the "Please update before proceeding".   The message that I added come from the ticket .  https://tools.hmcts.net/jira/browse/RDM-2676.

Reply:
Mario I ended up removing the "  Please update before proceeding".  As suggested . We need to update the ticket too.